packages/deployer/src/deployment.js

```diff
@@ -161,64 +163,36 @@ class Deployment {
   async _preFlightCheck(contract) {
     // Check that contract is not array
     if (Array.isArray(contract)) {
-      const message = await this.emitter.emit("error", {
+      const data = {
         type: "noBatches",
-        contract: null
-      });
+        contract
+      };
+      let message;
+      if (this.options && this.options.events) {
+        message = await this.options.events.emit("deployment:error", data);
+      }
 
-      throw new Error(message);
+      throw new Error(sanitizeMessage(message));
     }
 
     // Check bytecode
     if (contract.bytecode === "0x") {
-      const message = await this.emitter.emit("error", {
+      const data = {
         type: "noBytecode",
-        contract: contract
-      });
+        contract
+      };
+      let message;
+      if (this.options && this.options.events) {
+        message = await this.options.events.emit("deployment:error", data);
+      }
 
-      throw new Error(message);
+      throw new Error(sanitizeMessage(message));
     }
 
     // Check network
     await contract.detectNetwork();
   }
 
-  /**
-   * Handler for contract's `transactionHash` event. Rebroadcasts as a deployer event
-   * @private
-   * @param  {Object} parent Deployment instance. Local `this` belongs to promievent
-   * @param  {String} hash   tranactionHash
-   */
-  async _hashCb(parent, state, hash) {
-    const eventArgs = {
-      contractName: state.contractName,
-      transactionHash: hash
-    };
-    state.transactionHash = hash;
```

Comment:
I think this was removed because as far as I can tell it is not used anywhere in the code. Please double-check me on this but I couldn't find a reference to this method anywhere.

Reply:
Hm, maybe keep it in and remove it as a separate PR, to make it easy to revert later if it breaks something?

I don't feel too strongly about it. Would be curious what was using it in the past (if anything)